src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/GenericsHashtableNode.cs

```diff
@@ -0,0 +1,87 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.IO;
+using System.Diagnostics;
+using System.Collections.Generic;
+
+using Internal.Text;
+using Internal.TypeSystem;
+using Internal.NativeFormat;
+
+namespace ILCompiler.DependencyAnalysis
+{
+    /// <summary>
+    /// Represents a hashtable of all compiled generic type instantiations
+    /// </summary>
+    internal sealed class GenericsHashtableNode : ObjectNode, ISymbolNode
+    {
+        private ObjectAndOffsetSymbolNode _endSymbol;
+        private ExternalReferencesTableNode _externalReferences;
+
+        private NativeWriter _writer;
+        private Section _tableSection;
+        private VertexHashtable _hashtable;
+
+        private HashSet<TypeDesc> _genericTypeInstantiations;
```

Comment:
We should do this bookkeeping in MetadataManager. We already keep a list of all types that generate an EEType in the current compilation in it (see `TypeGeneratesEEType` method). You could expose a new method on MetadataGeneration that gives you an IEnumerable of all the types with EETypes generated and just filer them out in `GetData` below. Having a bunch of places that do the same thing is not maintainable.